### ISSUE TITLE
Build failure: pitivi, python311Packages.librosa

### ISSUE THREAD

**Issue by me00001** [NixOS/nixpkgs#306859]
### Steps To Reproduce

1. Be in nixos-unstable chanale
2. Run `nix-shell -p pitivi`

### Build log
https://paste.dismail.de/?6a40f862893c155b#F4P875mfQBHBDCaepTDseEU9CRmfNhy9gbfh55tyGaSC
``` python
Running phase: pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.11.9, pytest-8.1.1, pluggy-1.4.0
Matplotlib: 3.8.3
Freetype: 2.13.2
rootdir: /build/source
configfile: setup.cfg
plugins: mpl-0.16.1
collected 12910 items / 1 error / 48 deselected / 12862 selected               

==================================== ERRORS ====================================
____________________ ERROR collecting tests/test_filters.py ____________________
tests/test_filters.py:486: in <module>
    "pre_win", [scipy.signal.hann(16), list(scipy.signal.hann(16)), [1, 1, 1]]
E   AttributeError: module 'scipy.signal' has no attribute 'hann'
=========================== short test summary info ============================
ERROR tests/test_filters.py - AttributeError: module 'scipy.signal' has no attribute 'hann'
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
================= 48 deselected, 4 warnings, 1 error in 16.00s =================
/nix/store/9wnvhjyxjykwn5y06xc9a2h8rs5fbfia-stdenv-linux/setup: line 1579: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/1ps9paiq7ivs210lgmnzqnnkhaz92agp-python3.11-librosa-0.10.1.drv' failed with exit code 2;
       last 10 log lines:
       > ==================================== ERRORS ====================================
       > ____________________ ERROR collecting tests/test_filters.py ____________________
       > tests/test_filters.py:486: in <module>
       >     "pre_win", [scipy.signal.hann(16), list(scipy.signal.hann(16)), [1, 1, 1]]
       > E   AttributeError: module 'scipy.signal' has no attribute 'hann'
       > =========================== short test summary info ============================
       > ERROR tests/test_filters.py - AttributeError: module 'scipy.signal' has no attribute 'hann'
       > !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
       > ================= 48 deselected, 4 warnings, 1 error in 16.00s =================
       > /nix/store/9wnvhjyxjykwn5y06xc9a2h8rs5fbfia-stdenv-linux/setup: line 1579: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix-store -l /nix/store/1ps9paiq7ivs210lgmnzqnnkhaz92agp-python3.11-librosa-0.10.1.drv'.
error: 1 dependencies of derivation '/nix/store/8bcip9c0504qwnhvx86b8qpcgj4yywv7-pitivi-2023.03.drv' failed to build

```

### Notify maintainers

@GuillaumeDesforges
@AkechiShiro

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.28, NixOS, 24.05 (Uakari), 24.05pre615930.572af610f615`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - channels(suliman): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by AkechiShiro** [comment]
Hello @me00001 I'm not able to reproduce this issue. Thing is, I'm not on the unstable channel like you but I've used this built script (running `nix-shell script.nix` in order to pull the latest unstable Nixpkgs sources :  
```nix
let
  unstableTarball =
    fetchTarball
      https://github.com/NixOS/nixpkgs-channels/archive/nixos-unstable.tar.gz;
  pkgs = import <nixpkgs> {}; 
  unstable = import unstableTarball {};

  shell = pkgs.mkShell {
    buildInputs = [ unstable.pitivi unstable.librsvg ];
  };  
in shell
```

Pitivi fails to launch but still the errors is different, it is not due to a build failure on my side.

**Comment by GuillaumeDesforges** [comment]
I think I could reproduce the error on the specific nixpkgs commit (572af610f615)

**Comment by AkechiShiro** [comment]
I did reproduce the error too, I'm sorry I was using something a very deprecated tarball in my previous comment.

**Comment by jerith666** [comment]
I found two reports of this upstream:

https://github.com/librosa/librosa/issues/1834
https://github.com/librosa/librosa/issues/1830

It's supposed to be fixed in `0.10.2`; https://github.com/librosa/librosa/pull/1832 looks like it tracks release of that version.

**Comment by lelgenio** [comment]
Fixed in https://github.com/NixOS/nixpkgs/pull/309391
